sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -357,3 +361,134 @@ def test_imputation_copy():
+        X2_in = retype(X2)
+        # features = "auto":
+        indicator = MissingIndicator(missing_values=missing_values,
+                                     sparse=sparse_param)
+        X1_tr = indicator.fit_transform(X1_in)
+        X2_tr = indicator.transform(X2_in)
+        features = indicator.features_
+        assert_array_equal(expect_feat_missing, features)
+        assert_type(type(X2_tr), sparse.issparse(X2_in), sparse_param,
+                    missing_values)
+        assert_type(type(X1_tr), sparse.issparse(X1_in), sparse_param,
+                    missing_values)
+        if len(features) != 0:
+            assert_mask(X2_tr, mask_X2[:, features])
+            assert_mask(X1_tr, mask_X1[:, features])
+        else:
+            assert_mask(X2_tr, mask_X2)
+            assert_mask(X1_tr, mask_X1)
+
+        # features = "all"
+        indicator = clone(indicator).set_params(features="all")
+        X1_tr = indicator.fit_transform(X1_in)
+        X2_tr = indicator.transform(X2_in)
+        features = np.arange(X2.shape[1])
+        assert_type(type(X1_tr), sparse.issparse(X1_in), sparse_param,
+                    missing_values)
+        assert_type(type(X2_tr), sparse.issparse(X2_in), sparse_param,
+                    missing_values)
+        assert_mask(X2_tr, mask_X2[:, features])
+        assert_mask(X1_tr, mask_X1[:, features])
+
+        features = [1, 2]
+        indicator = clone(indicator).set_params(features=features)
+        X1_tr = indicator.fit_transform(X1_in)
+        X2_tr = indicator.transform(X2_in)
+        assert_type(type(X2_tr), sparse.issparse(X2_in), sparse_param,
+                    missing_values)
+        assert_type(type(X1_tr), sparse.issparse(X1_in), sparse_param,
+                    missing_values)
+        assert_mask(X2_tr, mask_X2[:, features])
+        assert_mask(X1_tr, mask_X1[:, features])
+
+    for X1, X2, missing_values in [(X1_orig, X2_orig, np.nan),
+                                   (X1_orig + 1, X2_orig + 1, 0)]:
+        for retype in [lambda x: x.tolist(), np.array, sparse.csr_matrix,
+                       sparse.csc_matrix, sparse.lil_matrix]:
+            for sparse_param in [True, False, 'auto']:
+                _check_missing_indicator(X1, X2, retype, sparse_param,
+                                         missing_values)
+
+
+def test_missing_indicator_error():
+    X1 = np.array([
+          [-1,  1,  3],
+          [4,  0, -1],
+          [8,  1,  0]
+    ])
+    X2 = np.array([
+          [5, -1, -1],
+          [-1,  2,  3],
+          [2,  4,  0]
+    ])
+    indicator = MissingIndicator(missing_values=-1)
+    indicator.fit(X1)
+    missing_features_fit = np.sum(X1 == -1, axis=0).nonzero()[0]
+    missing_features_tr = np.where(np.any(X2 == -1, axis=0))[0]
+    extra_missing_features = np.setdiff1d(missing_features_tr,
+                                          missing_features_fit)
+    err_msg = ("The features \{0} have missing values "
```

Comment:
Removing this backslash is failing the assertion check.

Reply:
Right. Because you are subbing in an unescaped string and treating it as a regex. The string starts with [. So this looks weird. You could use `re.escape(extra_missing_features)` instead of `extra_missing_features`.

But really, I think you should just put `[1]` in and not have generic logic like the setdiff1d in the test: you know which features are the issue for the specific example.